src/librbd/AioObjectRequest.cc

```diff
@@ -410,27 +451,10 @@ namespace librbd {
           		       << m_object_off << "~" << m_object_len << dendl;
         m_state = LIBRBD_AIO_WRITE_PRE;
 
-        uint8_t new_state;
```

Comment:
We need the new state to decide what to do after updating the object map. WRITE_PRE is now unnecessary, though, so I'll remove it.